src/main/java/mate/academy/bookshop/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package mate.academy.bookshop.service.shoppingcart;
+
+import jakarta.transaction.Transactional;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.cartitem.CartItemRequestDto;
+import mate.academy.bookshop.dto.cartitem.CartItemUpdateDto;
+import mate.academy.bookshop.dto.shoppingcart.ShoppingCartResponseDto;
+import mate.academy.bookshop.exceptions.EntityNotFoundException;
+import mate.academy.bookshop.mapper.CartItemMapper;
+import mate.academy.bookshop.mapper.ShoppingCartMapper;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.model.User;
+import mate.academy.bookshop.repository.CartItemRepository;
+import mate.academy.bookshop.repository.ShoppingCartRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartResponseDto getShoppingCart(Long userId) {
+        ShoppingCart shoppingCartByUserId = getShoppingCartByUserId(userId);
+        return shoppingCartMapper.toDto(shoppingCartByUserId);
+    }
+
+    @Override
+    public ShoppingCartResponseDto addCartItem(CartItemRequestDto cartItemUpdateDto, Long userId) {
+        ShoppingCart shoppingCartByUserId = getShoppingCartByUserId(userId);
+
+        Optional<CartItem> existingCartItem = shoppingCartByUserId
+                .getCartItems()
+                .stream()
+                .filter(cartItem -> cartItem
+                        .getBook()
+                        .getId()
+                        .equals(cartItemUpdateDto.getBookId()))
+                .findFirst();
+
+        if (existingCartItem.isPresent()) {
+            CartItem cartItem = existingCartItem.get();
```

Comment:
It is recommended to use `orElseThrow()` instead of `get()'. You can also see this in the java doc
```suggestion
            CartItem cartItem = existingCartItem.orElseThrow();
```